### ISSUE TITLE
collab_learner gives KeyErrors when querying biases as collabdatabunch did not include all items and users in the training and the validation set split

### ISSUE THREAD

**Issue by hantusk** [fastai/fastai#1586]
When using CollabDataBunch and collab_learner, the automatically split training set does not include all items. This causes KeyErrors e.g. when trying to get the biases from the learner, when you request an item from your original dataset that was did not appear in the training set.

```text
=== Software === 
python       : 3.6.7
fastai       : 1.0.42
fastprogress : 0.1.18
torch        : 1.0.0
torch cuda   : 10.0.130 / is **Not available** 

=== Hardware === 
No GPUs available 

=== Environment === 
platform     : Linux-4.14.79+-x86_64-with-Ubuntu-18.04-bionic
distro       : #1 SMP Wed Dec 19 21:19:13 PST 2018
conda env    : Unknown
python       : /usr/bin/python3
sys.path     : 
/env/python
/usr/lib/python36.zip
/usr/lib/python3.6
/usr/lib/python3.6/lib-dynload
/usr/local/lib/python3.6/dist-packages
/usr/lib/python3/dist-packages
/usr/local/lib/python3.6/dist-packages/IPython/extensions
/root/.ipython
no supported gpus found on this system
```

** Reproduction **
No reproduction gist included.

**Expected behavior**
Instead of a KeyError, I would expect an error message saying item does not exist in the model, and that it might be caused by the dataset being split up into training and validation sets.

**Comment by hantusk** [comment]
You can probably tell I spent a bit of time debugging :)

You are welcome to close the issue, if providing hypotheses for why the user is experiencing an error is not a standard that you strive in the exception handling across the fastai library.

**Comment by sgugger** [comment]
Providing clear error messages is definitely something we want, hopefully this last commit should make it easier in your case (not entirely sure since you didn't provide a minimal example of the problem).

**Comment by hantusk** [comment]
Your patch looks perfect, and catches the exception where i got it as well. Issue fully resolved, thanks :)
